plugins/templates/dialogs/templates.js

```diff
@@ -59,29 +59,31 @@
 
 		// Insert the specified template content into editor.
 		// @param {Number} index
-		function insertTemplate( html ) {
+		function insertTemplate( htmlFile ) {
 			var dialog = CKEDITOR.dialog.getCurrent(),
 				isReplace = dialog.getValueOf( 'selectTpl', 'chkInsertOpt' );
 
-			if ( isReplace ) {
-				editor.fire( 'saveSnapshot' );
-				// Everything should happen after the document is loaded (https://dev.ckeditor.com/ticket/4073).
-				editor.setData( html, function() {
+			htmlFile.then(html => {
```

Comment:
Unfortunately this change is not backwards compatible. The new way of getting template should be added alongside the old one (so there should be two independent flows after the change).